diff -Naur bazel/gflags.bzl bazel/gflags.bzl
--- bazel/gflags.bzl	2017-11-03 05:21:22.000000000 -0700
+++ bazel/gflags.bzl	2017-12-09 18:21:31.588867780 -0800
@@ -65,7 +65,7 @@
         "-DHAVE_INTTYPES_H",
         "-DHAVE_SYS_STAT_H",
         "-DHAVE_UNISTD_H",
-        "-DHAVE_FNMATCH_H",
+        #"-DHAVE_FNMATCH_H",
         "-DHAVE_STRTOLL",
         "-DHAVE_STRTOQ",
         "-DHAVE_PTHREAD",
